site/src/pages/WorkspacePage/WorkspacePage.test.tsx

```diff
@@ -181,6 +183,7 @@ describe("Workspace Page", () => {
 
   describe("Resources", () => {
     it("shows the status of each agent in each resource", async () => {
+      const getTemplateMock = jest.spyOn(api, "getTemplate").mockResolvedValueOnce(MockTemplate)
```

Comment:
It's generally better to test things that are apparent to the user than internals like whether a function was called, but sometimes it's necessary. Is this one of the cases where testing user-apparent effects is too hard?

Reply:
I'm not sure  -- I'll try seeing if I can make some assertions about child components here.

Reply:
Actually, it seems like testing properties of child components is considered somewhat of an anti-pattern. I'll factor some more stuff out to a util file and test it separately.